### PR TITLE
Added BOM configuration parameter to RenderConfig.Csv

### PR DESCRIPTION
**Pull request by djspiewak**
Some destinations don't correctly handle the utf-8 byte order mark, so we need the ability to remove it.

### CODE REVIEW

**Comment by djspiewak** [comment]
Published as `143.1.0-e244a33`

File: api/src/main/scala/quasar/api/push/RenderConfig.scala [hunk 1]
```diff
@@ -23,8 +23,11 @@ import java.time.format.DateTimeFormatter
 object RenderConfig {
   import DateTimeFormatter._
 
+  // please note that binary compatibility is *only* guaranteed on this if you
+  // construct instances named arguments
```
Comment (jsantos17) [inline api/src/main/scala/quasar/api/push/RenderConfig.scala]: Not sure I follow. Would you mind explaining this comment?

Comment (djspiewak) [reply api/src/main/scala/quasar/api/push/RenderConfig.scala]: It means that we *don't* guarantee binary compatibility for something like this:

```scala
RenderConfig.Csv(false, format1, format2, format3, ...)
```

Basically named arguments here are being used as a way to ensure that this remains binary compatible even as we add to it, reducing the conflicts and allowing saner eviction.

Comment (alissapajer) [reply api/src/main/scala/quasar/api/push/RenderConfig.scala]: you forgot a word in the comment, would be more clear if it said

`construct instances using named arguments`

**Comment by alissapajer** [comment]
@djspiewak conflicts

**Pull request merged**